src/OpenLoco/Vehicles/VehicleManager.cpp

```diff
@@ -14,3 +15,47 @@ namespace OpenLoco::VehicleManager
         call(0x004C3A0C, regs);
     }
 }
+
+namespace OpenLoco::Vehicles::RoutingManager
+{
+    constexpr uint16_t allocated_but_free_routing_station = -2; // Indicates that this array is allocated to a vehicle but no station has been set.
+    constexpr uint16_t routingNull = -1;                        // Indicates that this array is allocated to a vehicle but no station has been set.
+
+    static auto& routings() { return getGameState().routings; }
+
+    static std::optional<uint16_t> findFreeRoutingVehicleRef()
+    {
+        const auto& routingArr = routings();
+        const auto res = std::find_if(std::begin(routingArr), std::end(routingArr), [](const auto& route) { return route[0] == routingNull; });
+        if (res == std::end(routingArr))
+        {
+            return std::nullopt;
+        }
+        return std::distance(std::begin(routingArr), res);
+    }
+
+    bool isEmptyRoutingSlotAvailable()
+    {
+        return findFreeRoutingVehicleRef().has_value();
+    }
+
+    // 0x004B1E00
+    std::optional<RoutingHandle> getAndAllocateFreeRoutingHandle()
+    {
+        auto vehicleRef = findFreeRoutingVehicleRef();
+        if (vehicleRef.has_value())
+        {
+            auto& vehRoutingArr = routings()[*vehicleRef];
+            std::fill(std::begin(vehRoutingArr), std::end(vehRoutingArr), allocated_but_free_routing_station);
+            return { RoutingHandle(*vehicleRef, 0) };
+        }
+        return std::nullopt;
+    }
+
+    // 0x004B1E77
+    void freeRoutingHandle(const RoutingHandle routing)
```

Comment:
freeVehicleRouting? need to make it clear this is clearing the whole routing for a vehicle not just the individual index.